sotodlib/site_pipeline/make_ml_map.py

```diff
@@ -0,0 +1,322 @@
+                     help="Path to mapmaker config.yaml file")
+
+    parser.add_argument("--query", type=str)
+    parser.add_argument("--freq", type=str)
+    parser.add_argument("--area", type=str)
+    parser.add_argument("--odir", type=str)
+    parser.add_argument("--prefix", type=str)
+    parser.add_argument('-C', "--comps",   type=str, help="T,Q, and/or U")
+    parser.add_argument("-c", "--context", type=str,  help="Context containing TODs")
+    parser.add_argument("-n", "--ntod",    type=int)
+    parser.add_argument(      "--tods",    type=str)
+    parser.add_argument(      "--nset",    type=int)
+    parser.add_argument("-N", "--nmat",    type=str)
+    parser.add_argument(      "--max-dets",type=int)
+    parser.add_argument("-S", "--site",    type=str)
+    parser.add_argument("-v", "--verbose", action="count")
+    parser.add_argument("-q", "--quiet",   action="count")
+    parser.add_argument("-@", "--center-at", type=str)
+    parser.add_argument("-w", "--window",  type=float)
+    parser.add_argument("-i", "--inject",  type=str)
+    parser.add_argument(      "--nocal",   action="store_true", default=True)
+    parser.add_argument(      "--nmat-dir", type=str)
+    parser.add_argument(      "--nmat-mode", type=str, help="How to build the noise matrix. 'build': Always build from tod. 'cache': Use if available in nmat-dir, otherwise build and save. 'load': Load from nmat-dir, error if missing. 'save': Build from tod and save.")
+    parser.add_argument("-d", "--downsample", type=int, help="Downsample TOD by this factor")
+    parser.add_argument(      "--maxiter",    type=int)
+    parser.add_argument("-T", "--tiled"  ,    type=int)
+    parser.add_argument("-W", "--wafer"  ,   type=str, nargs='+', help="Detector wafer subset to map with")
+    return parser
+
+
+def _get_config(config_file):
+    return yaml.safe_load(open(config_file,'r'))
+
+
+
+def main(config_file=None, defaults=defaults, **args):
+    
+    cfg = defaults
+    if config_file is not None:
+        # Load in config file
+        cfg_from_file = _get_config(config_file)
+        
+        # Check which fields have been supplied in config.yaml
+        for field, file_val in cfg_from_file.items():
+            # If an optional arg is supplied, in the config file
+            # add it here
+            if field not in cfg.keys():
+                cfg[field] = cfg_from_file[field]
+            # If the value in the config file is different from the 
+            # default, update it
+            elif cfg[field] != file_val:
+                cfg[field] = file_val
+    else:
+        print("No config file provided, assuming default values") 
+
+    # Merge flags from config file and defaults with any passed through CLI
+    for key, item in args.items():
+        if item is not None:
+            cfg[key] = item
+    # Certain fields are required. Check if they are all supplied here
+    required_fields = ['query','freq','area','odir','prefix','context']
```

Comment:
Reasonable default query is "1" (all tods).  I think odir and prefix could have reasonable defaults too ("output", "").

Reply:
I think that makes sense. Question about area: right now I run your `get_geom` script to generate a an `area` from a `context.yaml`. Where do we anticipate this step will happen when we move to `site-pipeline`? In principle, that could be added to the map-making module, but maybe it needs to be separate for some reason?

Reply:
The library level code can definitely live in mapmaking module, or perhaps in coords.helpers.

There some argument that make-ml-map should be able to generate this file itself.  It could be configurable like  the noise matrix args; with the option of computing it (on the fly as part of the preprocessing), or loading a saved one from disk.  We probably want a stand-alone way to generate this file though, which does the minimal thing needed.  That could be a short site-pipeline script (make-footprint-file?), or just a special sub-mode of make-ml-map.